### DISCUSSION TITLE
关于POST/GET请求分流的问题

### DISCUSSION THREAD

**Discussion by sigezhao** [bfenetworks/bfe#1034] [Q&A (问题和回答)] [answered]
假定按照用户ID分流：
GET请求
req_query_value_hash_in(key, hash_value_list, case_insensitive)，从文档里面，看到该hash_value_list参数值多个之间使用‘|’（模值0～9999）,那将0～9999分成两份，那 hash_value_list 该怎么写呢？

POST，请求的ID在POST data里面，从文档里面没找到相关的原语实现

期待官方的开发人员解惑

**Comment by iyangsj** [comment]
>> 将0～9999分成两份，那 hash_value_list 该怎么写呢？

基本的用法如下：
- 不连续的模值可以使用|，例如：1或者10写为"1|10"
- 连续的模值范围可以使用-，例如从0到50写为 "0-50";
- 这两个方式也可以组合，例如1或 10到20 或100写为："1|10-20|100"

对应上述需求的示例写法:
req_query_value_hash_in("userid", "0-4999", false)
req_query_value_hash_in("userid", "5000-9999", false)

**Comment by iyangsj** [comment, marked as answer]
>> POST，请求的ID在POST data里面，从文档里面没找到相关的原语实现

由于请求body格式未知，无法直接支持。

建议做法：
- 编写扩展模块，解析请求body并将ID值设置到请求头部（例如query中）
- 基于如上类似方式再配置规则
